app_test.go

```diff
@@ -1263,7 +1262,6 @@ func TestAppStart(t *testing.T) {
 						return nil
 					},
 					OnStop: func(context.Context) error {
-						secondStop = true
 						time.Sleep(10 * time.Millisecond)
```

Comment:
I thought this hook was never supposed to run. Does this hook run if the machine is too slow? (and the fix is to just not care about this hook any more?)

Reply:
so Stop() does not have a timeout here. It can run if the test doesn't finish running before the firstStop hook.

Reply:
I thought stop hooks run in reverse order. So I guess if the second startHook finishes running some time in between Start() timing out and Stop() being called, the second stopHook will run.